src/client-services/github.client-service.test.tsx

```diff
@@ -22,10 +22,12 @@ describe('Github client service test', () => {
 
   beforeEach(() => {
     jest.clearAllMocks();
+    process.env.GITHUB_API_KEY = 'testToken';
```

Comment:
https://stackoverflow.com/questions/48033841/test-process-env-with-jest
see the accepted solution as a possible solution

what you did here is not ideal because you are creating side effects that might impact other tests later on. 

basically let's imagine  initially ..
process.env.A = 'a' <- and that is the initially state of env vars, which means all tests see process.env.A = 'a',
what you are doing in the test is manipulating the global environment state of process.env.A = 'b', then removing process.env.A after. 
which would result in change of global state. -> this would lead to bugs and unpredictable behavior if your other tests also depends on process.env.

Reply:
alternative to having problems with fetching env var is to simply have a function instead and mock the function. 
i.e.

getEnvVar(env: string): {
   return process.env['env']
}
then in test you never need to mock process.env again, just mock out getEnvVar function which is straight forward in jest.

Reply:
> https://stackoverflow.com/questions/48033841/test-process-env-with-jest see the accepted solution as a possible solution
> 
> what you did here is not ideal because you are creating side effects that might impact other tests later on.
> 
> basically let's imagine initially .. process.env.A = 'a' <- and that is the initially state of env vars, which means all tests see process.env.A = 'a', what you are doing in the test is manipulating the global environment state of process.env.A = 'b', then removing process.env.A after. which would result in change of global state. -> this would lead to bugs and unpredictable behavior if your other tests also depends on process.env.

You're right, thanks for showing me in details. I will make sure to only mock process.env for each test, not change global value